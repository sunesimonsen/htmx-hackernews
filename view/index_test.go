package view

import (
	"testing"

	"github.com/alecthomas/assert/v2"
	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/sunesimonsen/htmx-hackernews/mock"
	"github.com/sunesimonsen/htmx-hackernews/templates"
)

func TestIndexView(t *testing.T) {
	renderer := templates.NewRenderer()

	t.Run("when the view renders succesfully", func(t *testing.T) {
		view := IndexView{Templates: renderer}

		data, err := view.Render(mock.Params{}, mock.Headers{})

		assert.NoError(t, err)
		snaps.MatchSnapshot(t, string(data))
	})
}
