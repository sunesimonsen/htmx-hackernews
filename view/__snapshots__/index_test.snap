
[TestIndexView/when_the_view_renders_succesfully - 1]

<html lang="en">
<head>
    <title>Hackernews</title>
    <meta charset="UTF-8"/>
    <meta name="description" content="Hackernews example app in htmx and Go" lang="en"/>
    <link rel="icon" href="/assets/favicon.ico" type="image/x-icon">
    <script src="https://unpkg.com/htmx.org@1.9.4" integrity="sha384-zUfuhFKKZCbHTY6aRR46gxiqszMk5tcHjsVFxnUo8VMus4kHGVdIYVbOYYNlKmHV" crossorigin="anonymous"></script>
    <script src="/assets/scripts.js"></script>
    <link rel="stylesheet" href="/assets/main.css">
</head>
<body>
  <main>
    
<header class="topBar">
  
<a class="logo" href="/">
    <img src="/assets/y18.gif" />
    <span class="logo-caption">Hacker News</span>
</a>

</header>

    <section class="content">

<div class="stories" hx-get="/topstories?embed" hx-trigger="load">
  
  <article class="htmx-indicator story placeholder">
    <a class="title"><div class="skeleton" style="width: 60%"></div></a>
    <footer><div class="skeleton" style="width: 40%"></div></footer>
  </article>

  
  <article class="htmx-indicator story placeholder">
    <a class="title"><div class="skeleton" style="width: 60%"></div></a>
    <footer><div class="skeleton" style="width: 40%"></div></footer>
  </article>

  
  <article class="htmx-indicator story placeholder">
    <a class="title"><div class="skeleton" style="width: 60%"></div></a>
    <footer><div class="skeleton" style="width: 40%"></div></footer>
  </article>

  
  <article class="htmx-indicator story placeholder">
    <a class="title"><div class="skeleton" style="width: 60%"></div></a>
    <footer><div class="skeleton" style="width: 40%"></div></footer>
  </article>

  
  <article class="htmx-indicator story placeholder">
    <a class="title"><div class="skeleton" style="width: 60%"></div></a>
    <footer><div class="skeleton" style="width: 40%"></div></footer>
  </article>

</div>

    </section>
  </main>
</body>
</html>


---
